pkg/network/node/pod.go

```diff
@@ -288,6 +289,8 @@ func (m *podManager) processRequest(request *cniserver.PodRequest) *cniserver.Po
 		}
 		result.Err = err
 	case cniserver.CNI_DEL:
+		m.runningPodsLock.Lock()
+		defer m.runningPodsLock.Unlock()
```

Comment:
for consistency should we drop the lock before calling `teardown`?

Reply:
definitely, good catch.

Reply:
fixed.